### ISSUE TITLE
Function to more easily add Properties

### ISSUE THREAD

**Issue by jtc42** [labthings/python-labthings#10]
Something like an `attr_to_property` function.

You pass an object, and a string of one of its attributes eg (microscope, "name") just like Pythons `getattr`

Function then creates a view using get and set attr functions in the GET and POST methods, or if you passed a dictionary, not an object, then it'll handle that.

Add extra data with kwargs and you have fast, documented (use variable to schema converter) single value properties

**Comment by jtc42** [comment]
A very quick draft of how this might be achieved:

```
import uuid


class BaseResource:
    def __init__(self):
        super().__init__()

        self.func_map = {
            "get": self.get,
            "post": self.post,
            "put": self.put,
            "delete": self.delete,
        }

    def post(self):
        pass

    def put(self):
        pass

    def delete(self):
        pass

    def get(self):
        pass


# TODO: Turn into ABC
class BaseDynamicResource(BaseResource):
    def __init__(self):
        super().__init__()

        self.func_map = {
            "get": self.getter_func,
            "post": self.setter_func,
            "put": self.updater_func,
            "delete": self.deleter_func,
        }

        for method, func in self.func_map.items():
            if func:
                setattr(self, method, func)



def gen_view(
    getter_func=None,
    setter_func=None,
    updater_func=None,
    deleter_func=None
):
    """
    Create a subclass dynamically 
    """
    return type(
        f"GenCls{uuid.uuid4().hex}",
        (BaseDynamicResource, object),
        {
            "getter_func": getter_func,
            "setter_func": setter_func,
            "updater_func": updater_func,
            "deleter_func": deleter_func
        },
    )


def gen_property(
    get_property_function,
    setter_func=None,
    updater_func=None,
    deleter_func=None
):
    """
    Quickly create a property from a single get function.
    Wraps gen_view
    """
    return gen_view(
        getter_func=get_property_function,
        setter_func=setter_func,
        updater_func=updater_func,
        deleter_func=deleter_func,
    )


def gen_action(
    action_function,
    getter_func=None,
    updater_func=None,
    deleter_func=None,
):
    """
    Quickly create an action from a single action function.
    Wraps gen_view
    """
    return gen_view(
        getter_func=getter_func,
        setter_func=action_function,
        updater_func=updater_func,
        deleter_func=deleter_func,
    )


### Use these things in an example

def test_getter(view):
    return f"Hello there!"

print("Creating view class")
ViewClass = gen_view(getter_func=test_getter)
```

We should really try and do something where we use `getattr` and `setattr` so you can easily build a Property view that directly maps to some object attribute.

**Comment by jtc42** [comment]
Closed by 20361578e627689aad46db878a0ed31f909288ee
